entity/src/qualified_purl.rs

```diff
@@ -1,5 +1,141 @@
 use sea_orm::{entity::prelude::*, FromJsonQueryResult, FromQueryResult};
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_json::{json, Value};
 use std::collections::BTreeMap;
+use std::fmt::{Display, Formatter};
+use trustify_common::purl::Purl;
+
+// TODO: some day we might 'collapse' all the purl structs to this one, that day is not today.
+//
+// Instead of using composition or directly relating to any existing purl struct we will implement
+// to & from impl and in places we may invalidate DRY principle by copying across required code blocks.
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, FromJsonQueryResult)]
+pub struct CanonicalPurl {
```

Comment:
I suspect this could replace (or be replaced by?) `trustify_common::purl::Purl`.

Reply:
(non-blocking for this PR; iterate!)